mount/upgrader.go

```diff
@@ -43,11 +47,32 @@ func Upgrade(underlying Mount, rootdir, initial string) (*Upgrader, error) {
 		return ret, nil
 	}
 
-	if initial != "" {
-		if _, err := os.Stat(initial); err == nil {
-			ret.transient = initial
-			return ret, nil
-		}
+	if _, err := os.Stat(ret.transientFilePath()); err == nil {
+		return ret, nil
+	}
+	// clean up the existing transient if it's gone bad
+	_ = os.Remove(ret.transientFilePath())
+
+	if initial == "" {
+		return ret, nil
+	}
+	// we don't want to rename the file given by the client -> simply copy it and use the upgrader's file naming scheme here.
```

Comment:
We don't need to rename anything, and we absolutely don't want to duplicate the file, as that is wasteful. We should just save the initial transient path as provided by the client, even if it's outside the transient root.